lib/PipelineC/PipelineCTraceTool.cpp

```diff
@@ -0,0 +1,104 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/Support/Base64.h"
+
+#include "revng/ADT/ConstexprString.h"
+#include "revng/PipelineC/PipelineC.h"
+#include "revng/PipelineC/PipelineCTraceTool.h"
+#include "revng/PipelineC/TraceParsing.h"
+#include "revng/PipelineC/TracingCommon.h"
+
+static llvm::StringMap<std::set<size_t>> BufferRegistry;
+
+template<ConstexprString Name, typename ArgTupleT, size_t I = 0>
+static void handleArguments() {
+  using ArgT = std::tuple_element_t<I, ArgTupleT>;
+  constexpr int LH = LengthHint<Name, I>;
+  if constexpr (anyOf<ArgT, char *, const char *>() && LH >= 0) {
+    std::string StrName(Name);
+    BufferRegistry[StrName].insert(I);
+  }
+
+  if constexpr (I + 1 < std::tuple_size_v<ArgTupleT>)
+    handleArguments<Name, ArgTupleT, I + 1>();
+}
+
+template<ConstexprString Name, typename ReturnT, typename... Args>
+static void handleFunction(std::function<ReturnT(Args...)> Function) {
+  if constexpr (sizeof...(Args) > 0)
+    handleArguments<Name, std::tuple<Args...>>();
+}
+
+template<ConstexprString Name, typename FunctionT>
+static void registerFunction(FunctionT Function) {
+  handleFunction<Name>(std::function{ Function });
+}
+
+static Initializer init([]() {
+// Add autogenerated registerRunner calls to register the
+// other functions automatically
+#include "PipelineCTraceRunner.inc"
+});
```

Comment:
I'm not a fan of this pattern, I'd like to see the `Initializer` class go away.
Can't you simply do this in a constructor of a global object?

```c++
class Lol {
  Lol() {
#include "PipelineCTraceRunner.inc"
  }
} Initialize;

Reply:
>  Can't you simply do this in a constructor of a global object?

The advantage of `Initializer` was that it delayed initialization on the first call, maybe a premature optimization

Dropped `Initializer`, switched to a IIFE